maxim/tests/test_maxim_core_simple.py

```
@@ -0,0 +1,246 @@
+from maxim import Maxim, Config
+from maxim.cache import MaximCache, MaximInMemoryCache
+
+
+class FalsyCache(MaximCache):
+    """A cache that evaluates to False but is not None - used to test the 'cache is not None' fix."""
+
+    def __init__(self):
+        self.data = {}
+        self.was_used = False
+
+    def __bool__(self):
```

Ruff (0.11.9)

Missing return type annotation for special method `__bool__`

Add return type annotation: `bool`

(ANN204)